### DISCUSSION TITLE
WiFi won't connect if CCMP-256 or GCMP-256 enabled on network

### DISCUSSION THREAD

**Discussion by Ionic** [arendst/Tasmota#20121] [General]
My new Nous A1T device failed to connect to my WiFi network.

The symptoms were quite odd: it tried to connect, went through the authenticating phase (which is more like a no-op for WPA2-PSK/Personal), then disconnected without ever even trying to associate.

It took me hours to debug this issue, but eventually I figured out that enabling CCMP-256 (00-0f-ac:10) or GCMP-256 (00-0f-ac:9) leads to the device not being able to connect at all.

Disabling both ciphers immediately let it connect - all other settings unchanged. I eventually resorted to creating a new virtual AP/SSID with `{C,G}CMP-256` disabled just for the smart plug.

I'll probably create a PR for the FAQ to include this crucial bit of information at a later time, but, my general question is if that's a WiFi firmware or a Tasmota issue? It's difficult to debug this, although I guess I could just try to toggle CCMP-256 support twice and then fetch the logs after it connected successfully again. If it's "just" a Tasmota issue, this could probably be fixed, but I've seen a lot of bug reports that were just closed referencing "firmware issue, won't/can't fix in software".

Crucially, the weird part is that even though `{C,G}CMP-256` were enabled, the normal `{C,G}CMP-128` ciphers were also enabled at the same time, so the device SHOULD have been able to ignore the unknown ciphers and just use `CCMP-128` to connect. It doesn't do so, though, which sounds like a fixable bug.

**Comment by Jason2866** [comment]
Welcome in the hell of the closed source wifi stack provided from espressif.

**Reply by sfromis** [reply]
In general, it works best if you "dumb it down" to not using newfangled features from the last few years.

**Comment by mybabysexy** [comment]
I got this problem too, but figured that out before seeing this discussion. I guess that is firmware related problem.

Router / AP: Mikrotik hAP ax³
Tasmota 13.3.0 running on following devices:
- ESP32-C3 Super Mini
- ESP32 Dev Kit (DK)
- Wemos Mini D1 (ESP8266)

Test 1: CCMP + GCMP
All 3 of them worked

Test 2: CCMP + GCMP + CCMP 256
All 3 of them worked

Test 3: CCMP + GCMP + **GCMP 256**
- C3: AP timeout (won't connect)
- DK: worked
- D1: worked

Test 4: CCMP + GCMP + CCMP 256 + **GCMP 256**
- C3: AP timeout (won't connect)
- DK: worked
- D1: worked

Conclusion:
-> ESP32-C3 won't connect to my AP with **GCMP 256 enabled**
FW: `esp32c3-api1-20210207`

Bonus: my working config
![image](https://github.com/arendst/Tasmota/assets/20596043/48f99f72-ae5f-48ca-b773-b76184d652af)

**Reply by Ionic** [reply]
Thanks for sharing.

It's interesting that ESP8266 fails to connect if I enable CCMP-256 for me, while it did work for you. I haven't tested with 13.3.0 yet, though, I was on 13.2.0. Might need to retest with 13.3.0.

**Comment by Jason2866** [comment]
@mybabysexy @Ionic  Try Arduino core 3.0.0 too. It has changed closed source wifi libs. The closed source wifi libs do change often. A reproducible compare is only possible when a Tasmota version with same Arduino core is used. Look for the time stamp after Arduino/IDF version. That's the date of the IDF version used when the Tasmota Arduino framework was built.
